setup.py

```diff
@@ -3,31 +3,24 @@
 import ctypes
 import sys
 import unittest
+
 try:
     from setuptools import setup, Extension, Command
 except ImportError:
     from distutils.core import setup, Extension, Command
 from distutils.command.build_ext import build_ext
 from distutils.command.build_py import build_py
-from distutils.errors import CCompilerError, DistutilsExecError, \
-                             DistutilsPlatformError
-from pike import __version__
+from distutils.errors import CCompilerError, DistutilsExecError, DistutilsPlatformError
 
 # attempt building the kerberos extension
 try_krb = True
+ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError)
 
-if sys.platform == 'win32' and sys.version_info > (2, 6):
-   # 2.6's distutils.msvc9compiler can raise an IOError when failing to
-   # find the compiler
-   # It can also raise ValueError http://bugs.python.org/issue7511
-   ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError,
-                 IOError, ValueError)
-else:
-   ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError)
 
 class BuildFailed(Exception):
     pass
 
+
 class ve_build_ext(build_ext):
     # This class allows C extension building to fail.
```

Comment:
the kerb module is a value add, the package can still work even if the dependencies are not available on the system that is building pike.

that said, this whole weird, non-standard setup tricks leaves a pretty bad taste. i _think_ it would be ideal to build the pykerb module as an extra instead.